### ISSUE TITLE
Sort UI Bug with large number of mapping versions

### ISSUE THREAD

**Issue by jamlung-ri** [OpenConceptLab/ocl_issues#1668]
In a [concept on Staging](https://app.staging.openconceptlab.org/#/orgs/PIH/sources/PIH/concepts/7935/) with a large number of mappings, we were trying to update the sort weight for a single concept. However, OCL suggested that it would make 132 new mapping versions instead of just two. It looks like OCL changed all of the sort weights, rather than just the two mapping sort weights that are highlighted. Is the behavior expected?

Note: We should only deal with this is it's a quick fix.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBaGkwIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--6e2bc07aca67890f33832f8edd58f50a5b99e152/image%20%283%29.png)

**Comment by snyaggarwal** [comment]
@jamlung-ri Yes this was expected and I think we did discuss this. We need to apply sort_weight on all mappings to fix the order in a certain way. There may be some cases where we dont need to, but we took a call at the time to apply always.

**Comment by lnball** [comment]
Your comments don't clarify this.  If I add a new member to a large set, it didn't get a sort weight.  Is this fixed?  When I added a new member, I assumed it would be sorted to the bottom by default.  Not sure but I think OpenMRS expects a sort weight on set members (not NULL).

**Comment by jamlung-ri** [comment]
@lnball When you add a new member to a set, it won't get a sort weight until you have manually dragged the mappings into place. Otherwise, they won't have a sort weight and will have that Caution icon next to them. 

OCL currently doesn't auto-generate sort weights until there is some input from the user. For the time being, I might recommend adding any new members into the set, and then just apply to the sort weight to all of them at once. But in the future, I think we will want to make sure we have a setting that let's the user tell OCL to auto-assign sort weights, just to simplify this process as you're suggesting.
